src/compiler/docs/readme/markdown-dependencies.ts

```diff
@@ -29,6 +34,21 @@ export const depsToMarkdown = (cmp: d.JsonDocsComponent, cmps: d.JsonDocsCompone
     content.push(``);
   }
 
+  const { background: defaultBackground, textColor: defaultTextColor } = DEFAULT_TARGET_COMPONENT_STYLES;
+
+  let { background = defaultBackground, textColor = defaultTextColor } =
+    config.docs?.markdown?.targetComponent ?? DEFAULT_TARGET_COMPONENT_STYLES;
+
+  if (!isHexColor(background)) {
+    background = defaultBackground;
+    config.logger.warn('Default value for docs.markdown.targetComponent.background is being used.');
+  }
+
+  if (!isHexColor(textColor)) {
+    textColor = defaultTextColor;
+    config.logger.warn('Default value for docs.markdown.targetComponent.textColor is being used.');
+  }
```

Comment:
Alright, I may have mislead you here, so I apologize. When testing the warning I realized that this method executes for every component with dependencies, so it's a bit annoying that the warning logs for every instance. So, that made me realize we should probably move this logic to the [validateConfig](https://github.com/ionic-team/stencil/blob/main/src/compiler/config/validate-config.ts#L68) method (and update appropriate tests). That way, by this point, we know that the value from the config is valid and the warning will only log once per build. 

Again, very sorry I didn't catch this on the first review 😞

Reply:
No prob @tanner-reits
You are right, I didn't run the `build` command after the last changes... I could catch it too.
I'll have a look 🙂